py3status/modules/emerge_status.py

```diff
@@ -0,0 +1,144 @@
+        """
+        Use pgrep to check if emerge is running.
+
+        Returns true if at least one instance of emerge is running.
+        """
+        tmp = os.popen("pgrep emerge").read()
+        pcount = tmp.count("\n")
+        if pcount > 0:
+            return True
+        else:
+            return False
+
+    def _getProgress(self):
+        """
+        Get current progress of emerge.
+
+        returns a dict containing current and total value.
+        """
+        ret = {}
+        ret['current'] = None
+        ret['total'] = None
+        ret['err'] = False
+        ret['err_text'] = ""
+        ret['category'] = None
+        ret['pkg'] = None
+
+        input_data = []
+
+        """
+        Try to open file, if unable throw error to be catched later
+        """
+        try:
+            with open(self.emerge_log_file, 'r') as fp:
+                for line in fp:
+                    input_data.append(line)
+        except IOError as err:
+            ret['err'] = True
+            ret['err_text'] = "Opening {} failed: {}" \
+                .format(self.emerge_log_file, err)
+            return ret
+
+        """
+        Traverse emerge.log from bottom up to get latest information
+        """
+
+        input_data.reverse()
+
+        for line in input_data:
+            terminated = re.search("\*\*\* terminating\.", line)
+            if terminated:
+                ret['current'] = "0"
+                ret['total'] = "0"
+                break
+            else:
+                match = re.search(">>> emerge.*", line)
+                if match:
+                    status_re = (
+                        "\((?P<current>[0-9]+) of (?P<total>[0-9]+)\) "
+                        "(?P<category>[a-zA-Z0-9\-]+)\/(?P<pkg>[a-zA-Z0-9\.]+)"
+                    )
+                    res = re.search(status_re, match.group())
+                    if res:
+                        ret['current'] = res.group('current')
+                        ret['total'] = res.group('total')
+                        ret['category'] = res.group('category')
+                        ret['pkg'] = res.group('pkg')
+                    else:
+                        ret['err'] = True
+                        ret['err_text'] = 'Regex did not match a line'
+                    break
+        return ret
+
+    def emerge_status(self):
+        """
+        Emerge Status main routine
+        """
+        response = {}
+        response['cached_until'] = self.py3.time_in(self.cache_timeout)
+        if self._emergeRunning():
+            self.ret = self._getProgress()
+            if self.ret['err']:
+                response['full_text'] = self.py3.safe_format(
+                    u"emerge_status error: {err_text}", self.ret['err_text'])
+            else:
+                response['full_text'] = self.py3.safe_format(
+                    self.format, self.ret)
+        else:
+            if self.hide_if_zero:
+                response['full_text'] = ""
```

Comment:
I personally think this a rather confusing formatting style and more likely to discourage users from using a custom format. Even more since the function is not, yet, documented.

Reply:
I think @tobes may have a branch to deal with this stuff. What is more is that the formatters can do more than just hiding zeroes... IIRC it can do `\?if=max_length 1`, `\?if=min_length 1`, `\?not_zero`, `\?if=placeholder`, `{placeholder:d}` (decimal), `{placeholder:.0f}` (rounding a float), and can grow more later. See `./tests/test_formatter.py` for more possibilities. 

Maybe we will get it documented for `3.8` or `3.9`, but will be easier to fix this module in the `format` now rather than to try and fix it it later / deprecate.

Also, you can add `Examples: ` with `# hide if zero` comment too.